### ISSUE TITLE
Feature request: Feeder configurations for each job

### ISSUE THREAD

**Issue by sukarsa** [openpnp/openpnp#27]
It would be nice to put feeders in job file, so that each time we load the job the associated feeders are loaded as well.

**Comment by vonnieda** [comment]
More comments on this subject from @Laczen:

Hi Jason,

Regarding the feeders into jobs my idea is the following: when feeders are placed into jobs there coordinate reference system becomes the same as the PCB, so all feeders are referenced relatieve to the PCB location. So how would this work in practice:

A fixture is made that holds the PCB and the job feeders (a specific fixture for a job). The fixture is preparated separate from the machine (loaded with PCB and component). The fixture is loaded in the machine and using fiducials on the board or on the fixture the positioning (on the machine) is identified. The job processing  can then start. While the job is running a seconden fixture can be   prepared for a next run.

This wouldn't require extra functionality from the camera or the machine.

A typical use case would be a PCB that uses limited amounts of a specific component, or the high value components, or components that require a specific feeder that is not available. These components are loaded together with the PCB on the fixture.

Kind regards,

Jehudi

**Comment by vonnieda** [comment]
I had an idea in another thread ( https://groups.google.com/d/msgid/openpnp/933414d8-c57a-438f-a032-b594b6bfabd9%40googlegroups.com?utm_medium=email&utm_source=footer ) that I think would be very easily to implement and would satisfy this request without changing things significantly.

The idea is that we could add UI to the Feeders panel that allows you to move a feeder to/from the loaded job. Moving a feeder from the machine to the job would remove it from the machine configuration and add it to a new feeders section on the job. Then, whenever this job is loaded, the feeders in the job would be loaded into the machine and when the job is unloaded (or the program is exited) the feeders are removed. They could be marked as temporary as far as the machine is concerned, so that they can be easily cleaned up.

So, in summary:

* Add a new feeders element to the Job, which would just be a List<Feeder> just like that on Machine.
* Add a "Job?" checkbox to the Feeders table. When checked it moves the feeder from the machine to the job.
* Add a "temporary" flag to the Feeder class. When feeders are loaded from a job they are marked temporary. Whenever a job is loaded, unloaded or when the program is started, automatically delete any temporary feeders.
* When a job is loaded, copy the feeders in the job to the machine and set the "temporary" flag on them.

It will be necessary to keep the lists in sync with regards to the "Job?" checkbox. This can probably be done by id.

**Comment by vonnieda** [comment]
Given the lack of interest in this feature, and the difficulty of implementing it, I'm closing it. If it ever comes up again in the future we can consider it. In the meantime, I think ReferenceSlotAutoFeeder and Scripting are better answers to this.

**Comment by Laczen** [comment]
OK
